### PR TITLE
B-20564: Malformed Data Notification

### PR DESCRIPTION
**Pull request by TevinAdams**
Ticket: [B-20564](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20564)

TRDM could receive malformed TAC and LOA data and we will need to be notified when this happens. I built in implementation to collect the tacSysIds and loaSysIds of rows that have malformed data and include these in an email that is sent to us developers. To do this AWS SNS SDK was used. To allow this functionality to work I had to add  recommended AWS Root CAs to TRDMs truststore to allow the request to the SNS service. This created the issue of the truststore file being too large for its current storage method SSM. To solve this I stored the truststore file in S3 and adjusted TRDM to fetch its truststore from there. This will allow us to add certs without worrying about the file size.
After solving the truststore storage issue and adding recommended certs the HTTPS request to AWS SNS service was successful. I also removed the EmailService logic as that implementation is no longer needed.

This PR introduces new functionality such as:

1. SNS notification that sends an email to "Milmove-Developers" when malformed TAC and LOA data is detected while parsing the pipe delimited file of TGET data
2. TRDM Truststore file is now stored in S3, and is now being retrieved using S3 SDK

*Testing was done in the STG env with assistance from the Brightstars team.

- Screenshot of a successful request to AWS SNS to send a Malformed LOA message
SNS LOA Log

- Screenshot of a successful request to AWS SNS to send a Malformed TAC message
SNS TAC Log

- Screenshot of a successful request to retrieve TRDMs truststore from S3
S3 Truststore Fetch Log

- Screenshot of the keystore being successfully built and configured after fetching TRDMs truststore form S3
Keystore configured Log

-Screenshot of Malformed data notification (Note: this was done with test entries for tacSysIds because the issue didn't actually happen. Whenever real data is used and the malformed data is detected the tacSysIds are collected and that collection not being empty will trigger the notification. To test this I had to make that list not empty.)
Malformed Data MSG Sample

### CODE REVIEW

**Comment by TevinAdams** [comment]
Corrected typos in the description.

File: src/main/java/com/milmove/trdmlambda/milmove/service/SNSService.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            }
+
+            tacMsg += "\n" + tacSysIdsListHeader + "\n" + tacSysIdsListed;
+
+            send(engGovTopicARN, tacMsg);
+            logger.info("finished sending malformed TAC SNS");
+        } else {
+            logger.info("not sending malformed TAC SNS. List of malformed tacSysIds is empty");
+        }
+    }
```
Comment (cameroncaci) [inline src/main/java/com/milmove/trdmlambda/milmove/service/SNSService.java:66]: Recommending combining these two funcs as they're identify minus the param of "TAC" or "LOA" which can be a param

**Review by cameroncaci — CHANGES_REQUESTED** [review]
Needs a version bump

File: src/main/java/com/milmove/trdmlambda/milmove/service/SNSService.java (hunk 1 above)
Comment (TevinAdams) [reply src/main/java/com/milmove/trdmlambda/milmove/service/SNSService.java]: @cameroncaci  Just refactored to remove repetitive code

**Comment by TevinAdams** [comment]
> Corrected typos in the description.

@cameroncaci  Done

**Review by traskowskycaci — COMMENTED** [review]
Some changes in src/main/java/com/milmove/trdmlambda/milmove/util/Trdm.java were just spacing diffs where one line became two. Just curious why that spacing was changed?

**Comment by TevinAdams** [comment]
> Some changes in src/main/java/com/milmove/trdmlambda/milmove/util/Trdm.java were just spacing diffs where one line became two. Just curious why that spacing was changed?

@traskowskycaci I format all the files I touch with Java Redhat formatter

**Comment by traskowskycaci** [comment]
> > Some changes in src/main/java/com/milmove/trdmlambda/milmove/util/Trdm.java were just spacing diffs where one line became two. Just curious why that spacing was changed?
> 
> @traskowskycaci I format all the files I touch with Java Redhat formatter

Sounds good, I wondered if that was the case. Thanks for the explanation!

**Review by traskowskycaci — APPROVED** [review]
Makes sense to me overall, and I have seen those emails coming in as further proof. Looks good!

**Review by cameroncaci — APPROVED** [review]

**Pull request merged**